.github/workflows/unit-test.yml

```
@@ -50,7 +51,41 @@ jobs:
               run: npx lerna run build
 
             - name: Running the tests
-              run: npm run test:unit -- --ci --maxWorkers=${{ steps.cpu-cores.outputs.count }} --cacheDirectory="$HOME/.jest-cache"
+              run: |
+                  npm run test:unit -- \
+                    --ci \
+                    --maxWorkers="${{ steps.cpu-cores.outputs.count }}" \
+                    --shard="${{ matrix.shard }}" \
+                    --cacheDirectory="$HOME/.jest-cache"
+
+    unit-js-date:
+        name: JavaScript Date Tests (Node.js ${{ matrix.node }})
+        runs-on: ubuntu-latest
+        if: ${{ github.repository == 'WordPress/gutenberg' || github.event_name == 'pull_request' }}
+        strategy:
+            fail-fast: false
+            matrix:
+                node: ['20', '21']
+
+        steps:
+            - uses: actions/checkout@9bb56186c3b09b4f86b1c65136769dd318469633 # v4.1.2
+              with:
+                  show-progress: ${{ runner.debug == '1' && 'true' || 'false' }}
+
+            - name: Setup Node.js and install dependencies
+              uses: ./.github/setup-node
+              with:
+                  node-version: ${{ matrix.node }}
+
+            - name: Get number of CPU cores
+              uses: SimenB/github-actions-cpu-cores@97ba232459a8e02ff6121db9362b09661c875ab8 # v2.0.0
+              id: cpu-cores
+
+            - name: npm build
```

```suggestion
            - name: Run build scripts
```
Let's use a descriptive name of what the step represents instead of just saying generally what the command is.